lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -0,0 +1,454 @@
+    -- Usage of this function should be phased out.
+    genesisBlockNo = BlockNo 0
+
+
+--------------------------------------------------------------------------------
+--
+-- Network Parameters
+
+-- | Settings for configuring a MainNet network client
+mainnetVersionData
+    :: NodeVersionData
+mainnetVersionData =
+    ( NodeToClientVersionData
+        { networkMagic =
+            NetworkMagic $ fromIntegral $ W.getProtocolMagic W.mainnetMagic
+        }
+    , nodeToClientCodecCBORTerm
+    )
+
+-- | Settings for configuring a TestNet network client
+testnetVersionData
+    :: W.ProtocolMagic
+    -> NodeVersionData
+testnetVersionData pm =
+    ( NodeToClientVersionData
+        { networkMagic =
+            NetworkMagic $ fromIntegral $ W.getProtocolMagic pm
+        }
+    , nodeToClientCodecCBORTerm
+    )
+
+--------------------------------------------------------------------------------
+--
+-- Type Conversions
+
+-- fixme: maybe just toShelleyHash = ShelleyHash . CC.unsafeHashFromBytes
+toShelleyHash :: Crypto c => W.Hash "BlockHeader" -> (ShelleyHash c)
+toShelleyHash (W.Hash bytes) =
+    unsafeDeserialiseCbor fromCBOR
+    $ BL.fromStrict bytes
+
+toEpochSlots :: W.EpochLength -> EpochSlots
+toEpochSlots =
+    EpochSlots . fromIntegral . W.unEpochLength
+
+-- | Magic value for the absence of a block.
+hashOfNoParent :: W.Hash "BlockHeader"
+hashOfNoParent = W.Hash . BS.pack $ replicate 0 32
+
+toPoint
+    :: Crypto c
+    => W.Hash "Genesis"
+    -> W.EpochLength
+    -> W.BlockHeader
+    -> Point (ShelleyBlock c)
+toPoint genesisH epLength (W.BlockHeader sid _ h _)
+  | h == (coerce genesisH) = O.GenesisPoint
+  | otherwise = O.Point $ Point.block (toSlotNo epLength sid) (toShelleyHash h)
+
+toSlotNo :: W.EpochLength -> W.SlotId -> SlotNo
+toSlotNo epLength =
+    SlotNo . W.flatSlot epLength
+
+-- | SealedTx are the result of rightfully constructed byron transactions so, it
+-- is relatively safe to unserialize them from CBOR.
+toGenTx :: HasCallStack => W.SealedTx -> GenTx (ShelleyBlock c)
+toGenTx = error "todo: toGenTx not implemented"
+
+fromShelleyTx :: SL.Tx c -> W.Tx
+fromShelleyTx = error "todo: fromShelleyTx not implemented"
+
+fromShelleyBlock
+    :: Crypto c
+    => W.Hash "Genesis"
+    -> W.EpochLength
+    -> ShelleyBlock c
+    -> W.Block
+fromShelleyBlock _genesisHash epLength blk =
+    let
+       ShelleyBlock (SL.Block (SL.BHeader header _) txSeq) headerHash = blk
+       SL.TxSeq txs' = txSeq
+       txs = map fromShelleyTx $ toList txs'
+
+    in W.Block
+        { header = W.BlockHeader
+            { slotId =
+                fromSlotNo epLength $ SL.bheaderSlotNo header
+            , blockHeight =
+                fromBlockNo $ SL.bheaderBlockNo header
+            , headerHash = fromShelleyHash headerHash
+            , parentHeaderHash = fromPreviousHash hashOfNoParent $
+                SL.bheaderPrev header
+            }
+        , transactions = txs
+        , delegations  = []
```

Comment:
Is this always the case for now or are delegation certificates somewhat already specified in the ShelleyBlock type?

Reply:
Looks like we can access them

https://github.com/input-output-hk/cardano-ledger-specs/blob/caa0b377698aace2550d1a9948a050bfa296a3e0/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxData.hs#L315-L328